### ISSUE TITLE
MetacelloGoferPackage>>matches: incorrectly matches PetitIndent-JanKurs.11 for explicit package reference PetitIndent-JanKurs.11

### ISSUE THREAD

**Issue by dalehenrich** [dalehenrich/metacello-work#339]
see [this thread](http://forum.world.st/Wrong-version-loaded-using-Metacello-td4817478.html) for details

**Comment by dalehenrich** [comment]
From the emaill, the bug is in MetacelloGoferPackage>>matches:

```Smalltalk
matches: aLoadableReference
  "does Monticello-style #versionInfoFromVersionNamed: matching"

  | pFilename refFilename char |
  ((pFilename := self packageFilename) == nil
    or: [ self name = self packageFilename ])
    ifTrue: [ ^ super matches: aLoadableReference ].
  aLoadableReference name = pFilename
    ifTrue: [ ^ true ].
  (aLoadableReference name beginsWith: pFilename)
    ifFalse: [ ^ false ].
  refFilename := aLoadableReference metacelloPackageNameWithBranch at: 2.
  refFilename = pFilename
    ifTrue: [ ^ true ].
  pFilename size < refFilename size
    ifTrue: [ 
      (refFilename beginsWith: pFilename)
        ifFalse: [ ^ false ].
      (char := pFilename at: pFilename size) ~= $-
        ifTrue: [ char := refFilename at: pFilename size + 1 ] ]
    ifFalse: [ 
      (pFilename beginsWith: refFilename)
        ifFalse: [ ^ false ].
      (char := refFilename at: refFilename size) ~= $-
        ifTrue: [ char := pFilename at: refFilename size + 1 ] ].
  ^ char = $. or: [ char = $- ]
```

There are a set of test cases for this method in **MetacelloGoferPackageTestCase**, `testInstanceMatch01` through `testInstanceMatch06`

Basically we are not recognizing the difference between a fully specified file (in this case) and the a partially specified package name (allowed when doing `bleedingEdge` style loads) ... the `beginsWith:` calls are the culprits ...

**Comment by dalehenrich** [comment]
Fixed in PR #341
